rust-toolchain.toml

```diff
@@ -1,5 +1,5 @@
 [toolchain]
-channel = "nightly-2022-11-15"
+channel = "nightly-2023-05-22"
```

Comment:
I think we also need to update `.tool-versions`? (Although I dislike trying to convert between the nightly dated version and the semvers)

Reply:
Sounds like we should just scrap asdf for rust and remove the notes about the rust asdf from the readme.

Reply:
Double checking my conversation with Peter, he finds asdf for rust useful, (if rust is installed with rustup, asdf is a noop), so he asked to keep it up to date. .tool-versions is already on 1.69 (same as 0.9.43), so I think no update is needed. The readme could detail rustup and leave asdf as an exercise left to the reader.